src/namingService.ts

```diff
@@ -36,9 +38,10 @@ export default abstract class NamingService extends BaseConnection {
   abstract resolver(domain: string): Promise<string>;
   abstract chatId(domain: string): Promise<string>;
 
-  constructor(web3Provider?: Web3Provider) {
+  constructor(web3Provider?: Web3Provider, providerType?: ProviderType) {
```

Comment:
Provide the default value `"ethers"` here as well. Also we should mention it in the `Resolution` constructor documentation: if `web3Provider` is passed, it's treated as `"ethers"` provider if `providerType` is not specified.